docs/develop/smart-contracts/guidelines/message-delivery-guarantees.mdx

```diff
@@ -0,0 +1,70 @@
+import ConceptImage from '@site/src/components/conceptImage'
+
+# Message delivery guarantees
+
+TON is an asynchronous blockchain with complex structure which is very different from other blockchains. In this regard, new developers have questions about low-level things in TON. In this article we will have a look at one of these related to message delivery.
+
+## What is a message?
+
+If we look at Ethereum or almost any other synchronous blockchain, each transaction can contain several smart contract calls in it. For example, DEXs are performing multiple exchanges in one transaction if there are no liquidity for the selected trading pair.
+
+In asynchronous system you can't get response from the destination smart contract in the same transaction. Contract call may take a few blocks to be processed, depending on the length of the route between source and destination.
+
+To achieve the infinite sharding paradigm, it is necessary to ensure full parallelization, which means that the execution of transactions is independent of each other. Therefore, instead of transactions that affect and change the state of many contracts at a time in TON, each transaction is only executed on one smart contract and smart contracts are communicating through messages. That way, smart contracts can only interact with each other by calling their functions with special messages, and getting response from them in other messages later.
+
+If transaction in Ethereum is just a set of function calls in different contracts, transaction in TON is some inbound message that triggered it and some set of outbound messages which are sent to other contracts.
+
+## What is a Logical time?
+
+In such a system with asyncronous and parallel smart contract calls, it can be hard to define the order of actions to process. That's why each message in TON has it's _Logical time_ or _Lamport time_ (later just _lt_). It is used to understand which event caused another one and what do validator need to process first.
```

Comment:
In such a system with asyncronous and parallel smart contract calls, it can be hard to define the order of actions to process. That's why each message in TON has it's _Logical time_ or _Lamport time_ (later just _lt_). It is used to understand which event caused another and what **a** validator **needs** to process first.